### PR TITLE
[GSoC] Add option to import text files

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description
This just adds the option, not the actual text/csv import implementation. which will be the next PR

As the CSV importer will require the new backend, the options is only shown if the new backend is enabled

## Approach
On the commits

## How Has This Been Tested?

1. Enable the new backend
2. Tap the three dot menu on the Deck picker > Import
3. Choose "Text file (.txt, .csv)
4. Select a text file
5. See if the "CSV importer is not implemented yet" is shown

image

image

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt [hunk 1]
```diff
@@ -144,9 +144,18 @@ object ImportUtils {
...
+        fun isTextFile(fileName: String): Boolean {
+            return fileName.endsWith(".txt") || fileName.endsWith(".csv") || fileName.endsWith(".tsv")
+        }
+
         private fun handleContentProviderFile(context: Context, intent: Intent, dataList: ArrayList<Uri>): ImportResult {
             // Note: intent.getData() can be null. Use data instead.
 
+            val firstFile = getFileNameFromContentProvider(context, dataList[0])
+            if (firstFile != null && isTextFile(firstFile)) {
+                return ImportResult.fromErrorString("CSV importer isn't implemented yet")
```
Comment (krmanik) [inline AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt:156]: The string should be extracted to resources.

**Review by krmanik — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt (hunk 1 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt:156]: This is a temporary warning. I don't expect it to be here more than 1 week, so I hardcoded it to spare the translators' work. 

If there is a problem with this, I'd rather join this commit with the actual CSV importer PR. Whatever it's better for the reviewers.

**Review by krmanik — APPROVED** [review]

**Comment by BrayanDSO** [comment]
"Waiting for Feedback" because I've noticed that I have more work to do on `ImportUtils` than I expected

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt:156]: I think this 

> This is a temporary warning. I don't expect it to be here more than 1 week, so I hardcoded it to spare the translators' work.

is reasonable but also really appreciate Mani looking out for the strings

works for me

**Comment by BrayanDSO** [comment]
ImportUtils.kt could be way more cleaner, as it seems that it has got change on top of change and became somewhat messy. And there is https://github.com/ankidroid/Anki-Android/issues/12079, which is especially relevant. 

Currently my top priority is pushing the actual CSV importer, so I don't know when/if I am going to refactor ImportUtils

**Review by lukstbit — APPROVED** [review]

**Pull request merged**